### ISSUE TITLE
[BUG] - 2.0 NavbarMenu lacks focus trapping

### ISSUE THREAD

**Issue by rostero1** [heroui-inc/heroui#1298]
### Describe the bug

Elements beneath the NavbarMenu should not gain focus when the NavbarMenu is open.

### Steps to Reproduce the Bug or Issue

Setup Navbar with a NavbarMenu.  Ensure there are elements beneath the Navbar that can gain focus.  Shrink the page to so the hamburger menu is visible.  Click the hamburger menu to pull up the menu defined by NavbarMenu.  Hit Tab to change element focus.  You'll notice that elements behind the menu gain focus.

### Expected behavior

Only visible elements should gain focus.

### Operating System Version

macOS

### Browser

Chrome
